pkg/worker/model_status_update_batch_job.go

```diff
@@ -0,0 +1,86 @@
+package worker
+
+import (
+	"context"
+	"fmt"
+
+	faktory "github.com/contribsys/faktory/client"
+	faktory_worker "github.com/contribsys/faktory_worker_go"
+	"go.uber.org/zap"
+
+	"github.com/cmsgov/mint-app/pkg/logfields"
+	"github.com/cmsgov/mint-app/pkg/models"
+)
+
+var modelStatusUpdateJobMaxRetry = 2
+
+// ModelStatusUpdateBatchJob is the job the cron job initiates to check if models need a status update.
+// It will batch all child jobs, and when complete it will fire a callback
+func (w *Worker) ModelStatusUpdateBatchJob(ctx context.Context, args ...interface{}) error {
+	helper := faktory_worker.HelperFor(ctx)
+	// decorate the logger, but exclude the bid, the bid will be decorated when we create the batch
+	logger := loggerWithFaktoryFieldsWithoutBatchID(w.Logger, helper)
+	logger.Info("Getting collection of model plans that require status checking")
+
+	// TODO: Implement the logic to return the models to check? Or do we check every model plan?
+	modelsToUpdate, err := w.Store.ModelPlanCollection(logger, false)
```

Comment:
@OddTomBrooks, do you know if there are any models that shouldn't be checked? This code chunk currently only looks at active models, which makes sense to me. Are there any statuses we don't want to check for (so to save some faktory bandwidth)?

Reply:
There are certainly ways to optimize this query down, for instance we could make a SQL query that selects statuses where not in the excluded states. For instance, this logic could be excluded entirely by a specialized SQL query:

`
	// If the model plan is paused or canceled, we shouldn't suggest a new phase
	if modelStatus == models.ModelStatusPaused || modelStatus == models.ModelStatusCanceled {
		return nil, nil
	}
`

That being said, this is a pre-optimization. It's dubious if we would even have an improvement depending on caching behavior. More importantly, the dev cost to build this versus spending time elsewhere is probably not worth the investment. I suggest that if this becomes problematic in terms of performance as part of a profiling pass that we revisit it. Profiling is something I hope to get to later in this sprint if I have the capacity.

Reply:
That works for me, maybe it's worthwhile to note in the code that this could be revisited. I agree it's likely not a huge impact.